### DISCUSSION TITLE
Error when building 0.54.x (Visual Studio)

### DISCUSSION THREAD

**Discussion by Sevinfolds** [navidrome/navidrome#3584] [Q&A]
When building an application in visual studio, the following error appears

github.com/navidrome/navidrome/scanner/metadata/taglib
taglib_wrapper.cpp:6:10: fatal error: aifffile.h: There is no such file or directory
    6 | #include <aifffile.h>
    
  Maybe someone has encountered this? There are no such problems with versions 0.53.x.

**Comment by deluan** [comment]
There was some changes on how we depend on TagLib. 

Make sure you have the latest TagLib (>2.0.1) installed in your system. If you have and it is still not working, then another option is to download the same TagLib build we use from https://github.com/navidrome/cross-taglib/releases (make sure you download the correct version for your system) and set the env var `PKG_CONFIG_PATH` pointing to the `lib/pkgconfig` folder inside it. 

Ex: If you downloaded and unzipped it to `/User/johndoe/taglib`, set:
```
PKG_CONFIG_PATH=/User/johndoe/taglib/lib/pkgconfig
```

**Reply by Sevinfolds** [reply]
Thank you, everything worked out! But a new error has occurred.

github.com/navidrome/navidrome
./main.go:16:16: undefined: buildtags.NETGO

**Reply by deluan** [reply]
`netgo` is required to build Navidrome (see #700). I added code to force this, and avoid community packaging to leave it out and causing issues to users.

You can add build tags in VSCode and IntelliJ. In VSCode you add the following config to your `gopls` settings:
```
  "gopls": {
    "build.buildFlags": [
      "-tags=netgo"
    ]
  }
```

In IntelliJ, Go to Settings->Languages & Frameworks->Go->Build Tags:
![Screenshot 2024-12-24 at 9 23 29 AM](https://github.com/user-attachments/assets/4df0ef91-11e1-4d1d-8473-1bff59f53fb4)
